source/Mlos.Python/mlos/Optimizers/unit_tests/TestBayesianOptimizer.py

```diff
@@ -167,7 +193,14 @@ def test_bayesian_optimizer_on_simple_2d_quadratic_function_cold_start(self):
 
             bayesian_optimizer.register(input_values_df, target_values_df)
             if i > optimizer_config.min_samples_required_for_guided_design_of_experiments and i % 10 == 1:
-                print(f"[{i}/{num_iterations}] Optimum: {bayesian_optimizer.optimum()}")
+                _, all_targets = bayesian_optimizer.get_all_observations()
+                best_config, optimum = bayesian_optimizer.optimum()
+                print(f"[{i}/{num_iterations}] Optimum: {optimum}")
+                assert optimum.y == all_targets.min()[0]
+                assert input_space.contains_point(best_config)

```

Comment:
Is it just awkward to use self.assertTrue(), is it conflicting with the upcoming switch to pytest or is it just habit? Either way it will throw, but I'm curious if there are any reasons not to use the unittest's built-ins?

Reply:
Just saw your comment below.